references/detection/train_pytorch.py

```diff
@@ -243,17 +243,18 @@ def main(args):
         img_folder=os.path.join(args.train_path, 'images'),
         label_path=os.path.join(args.train_path, 'labels.json'),
         img_transforms=Compose(
-            ([T.Resize((args.input_size, args.input_size))] if not args.rotation else [])
-            + [
+            [
                 # Augmentations
                 T.RandomApply(T.ColorInversion(), .1),
                 ColorJitter(brightness=0.3, contrast=0.3, saturation=0.3, hue=0.02),
             ]
         ),
-        sample_transforms=T.SampleCompose([
-            T.RandomRotate(90, expand=True),
-            T.ImageTransform(T.Resize((args.input_size, args.input_size))),
-        ]) if args.rotation else None,
+        sample_transforms=T.SampleCompose(
+            [T.Resize((args.input_size, args.input_size), preserve_aspect_ratio=True)]
+            + ([T.RandomRotate(90, expand=True),
+                T.ImageTransform(T.Resize((args.input_size, args.input_size), preserve_aspect_ratio=True))
+                ] if args.rotation else [])
+        ),
```

Comment:
For the sake of reducing the amount of interpolation being done, perhaps we should start with the expanded rotation first then the resize

Reply:
Unfortunately with our GPU's size we need to resize before the rotation otherwise the interpolation on very big images make the training crash for big angles

Reply:
alright I thought that could be reason :sweat_smile: 
fair enough